adr/ADR-32.md

```diff
@@ -94,6 +94,10 @@ Services should respond to:
 All discovery and status responses contain the following fields:
 
 ```typescript
+    /**
+     * An identifier of the message type for example io.nats.micro.v1.stats
+     */
+    type: string,
```

Comment:
@aricart @ripienaar Why do we need this, we already have name, which is described as "kind". We don't need both, but maybe we just change `name` to `type`

Reply:
It's to identify some random piece of JSON that you might not exactly know where its from (you could for example publish these stats message into a stream regularly or to some remote service). 
It's for documentation. 
It's to help others wanting to write clients.  
It's to communicate design decisions.  
It's consistent with jetstream

unversioned json is just amateur hour

Reply:
We already have a generic data field in stats. Also, again, why would one instance of a service "foo" have a `type`, where another instance of "foo" have a different `type`. 
Recently it was requested that we add a `meta` data field to the info. Is this the same purpose? 
Also, why does the `type` need to be of every response ping/info/stats/schema

Reply:
Ideally the service backend can publish advisories:

 * At start: info
 * At stop: stats
 * Regularly: stats

Very commonly these kinds of advisories are picked up and routed to other services where things like what NATS subject it was obtained from is gone.  Then passed to things like azure/aws event gateways which route messages based on data from their content.

As it stands, there is no way to know what a message is if you are given just the message.

Reply:
> Why do we need this, we already have name, which is described as "kind". We don't need both, but maybe we just change `name` to `type`

one is an API type, the other is marking the data for processing by something that doesn't know what that response is. This give the possibility of other tooling like nats-cli to do something smart with that response